src/charm.py

```diff
@@ -375,32 +388,59 @@ def _render_template(self, filename, **kwargs):
     def _environment_issues(self) -> List[str]:
         """Check for environment issues and return a list of issues."""
         issues = []
+
         if values := _sysctl_get("net.ipv4.conf.all.rp_filter"):
             if values["net.ipv4.conf.all.rp_filter"] != "0":
                 log.warning(
                     "Disable rp_filter on Cilium interfaces since it may cause mangled packets to be dropped. (%s)",
                     "net.ipv4.conf.all.rp_filter=0",
                 )
                 issues.append("sysctl rp_filter enabled for interfaces.")
+
+        if self.stored.cilium_tunnel_protocol == "vxlan" and self._vxlan_port_already_in_use():
+            issues.append("vxlan dst port is already in use. Set another tunnel-port.")
+
         return issues
 
-    def _set_active_status(self):
-        if not self.stored.cilium_configured:
-            return
+    def _vxlan_port_already_in_use(self) -> bool:
+        log.info(f"checking for validity of vxlan tunnel on port {self.stored.cilium_tunnel_port}")
 
-        if self.model.config.get("enable-hubble") and not self.stored.hubble_configured:
-            return
+        ps_number_using_vxlan_dst_port = 0
 
-        if self.stored.unallowed_metrics:
-            return
+        with IPRoute() as ip:
+            links = ip.get_links()
 
+            for link in links:
+                attrs = dict(link["attrs"])
+                info_data = dict(attrs.get("IFLA_LINKINFO", {}).get("attrs", {})).get(
+                    "IFLA_INFO_DATA", {}
+                )
+                info_data_attrs = dict(info_data.get("attrs", {}))
+                if str(info_data_attrs.get("IFLA_VXLAN_PORT")) == self.stored.cilium_tunnel_port:
+                    log.info(
+                        f"interface {attrs.get('IFLA_IFNAME')} is using port {self.stored.cilium_tunnel_port}"
+                    )
+                    ps_number_using_vxlan_dst_port += 1
```

Comment:
Wow, the engineering in here is great for AI to help us think about in the future